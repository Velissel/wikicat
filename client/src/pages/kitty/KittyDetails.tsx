import React from 'react';
import { Badge, ListGroup, ListGroupItem } from 'react-bootstrap';

interface KittyDetailsProps {
  temperament: string;
  origin: string;
  lifeSpan: string;
  adaptability: number;
  affectionLevel: number;
  childFriendly: number;
  grooming: number;
  intelligence: number;
  healthIssues: number;
  socialNeeds: number;
  strangerFriendly: number;
}
export default function KittyDetails(props: KittyDetailsProps) {
  const { temperament, origin, lifeSpan, ...rest } = props;
  const numbers = [
    'adaptability',
    'affectionLevel',
    'childFriendly',
    'grooming',
    'intelligence',
    'healthIssues',
    'socialNeeds',
    'strangerFriendly',
  ];
  return (
    <ListGroup as="ol" data-testid="kitty-details">
      <ListGroupItem>
        <div className="ms-2 me-auto">
          <div className="fw-bold">temperament</div>
          {temperament}
        </div>
      </ListGroupItem>
      <ListGroupItem>
        <div className="ms-2 me-auto">
          <div className="fw-bold">origin</div>
          {origin}
        </div>
      </ListGroupItem>
      <ListGroupItem>
        <div className="ms-2 me-auto">
          <div className="fw-bold">life span</div>
          {lifeSpan}
        </div>
      </ListGroupItem>
      {numbers.map((key: keyof typeof rest) => {
        const value = rest[key];
        return (
          <ListGroupItem
            key={key}
            as="li"
            className="d-flex justify-content-between align-items-start"
          >
            <div className="ms-2 me-auto">
              <div className="fw-bold">{key}</div>
            </div>
            <Badge bg="primary" pill>
              {value}
            </Badge>
          </ListGroupItem>
        );
      })}
    </ListGroup>
  );
}
