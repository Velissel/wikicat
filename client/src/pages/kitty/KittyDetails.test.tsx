import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import KittyDetails from './KittyDetails';

describe('KittyDetails component', () => {
  it('should render', () => {
    const cat = {
      temperament: 'temp',
      origin: 'ori',
      lifeSpan: 'life',
      adaptability: 1,
      affectionLevel: 2,
      childFriendly: 3,
      grooming: 4,
      intelligence: 5,
      healthIssues: 6,
      socialNeeds: 7,
      strangerFriendly: 8,
    };
    const { container } = render(
      <BrowserRouter>
        <KittyDetails {...cat} />
      </BrowserRouter>,
    );

    Object.keys(cat).forEach((key) => {
      const value = cat[key];
      expect(
        screen.getByText(key === 'lifeSpan' ? 'life span' : key),
      ).toBeInTheDocument();
      expect(screen.getByText(value)).toBeInTheDocument();
    });
    expect(container).toMatchSnapshot();
  });
});
