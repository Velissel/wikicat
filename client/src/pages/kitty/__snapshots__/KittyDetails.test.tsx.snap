// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KittyDetails component should render 1`] = `
<div>
  <ol
    class="list-group"
  >
    <div
      class="list-group-item"
    >
      <div
        class="ms-2 me-auto"
      >
        <div
          class="fw-bold"
        >
          temperament
        </div>
        temp
      </div>
    </div>
    <div
      class="list-group-item"
    >
      <div
        class="ms-2 me-auto"
      >
        <div
          class="fw-bold"
        >
          origin
        </div>
        ori
      </div>
    </div>
    <div
      class="list-group-item"
    >
      <div
        class="ms-2 me-auto"
      >
        <div
          class="fw-bold"
        >
          life span
        </div>
        life
      </div>
    </div>
    <li
      class="d-flex justify-content-between align-items-start list-group-item"
    >
      <div
        class="ms-2 me-auto"
      >
        <div
          class="fw-bold"
        >
          adaptability
        </div>
      </div>
      <span
        class="badge rounded-pill bg-primary"
      >
        1
      </span>
    </li>
    <li
      class="d-flex justify-content-between align-items-start list-group-item"
    >
      <div
        class="ms-2 me-auto"
      >
        <div
          class="fw-bold"
        >
          affectionLevel
        </div>
      </div>
      <span
        class="badge rounded-pill bg-primary"
      >
        2
      </span>
    </li>
    <li
      class="d-flex justify-content-between align-items-start list-group-item"
    >
      <div
        class="ms-2 me-auto"
      >
        <div
          class="fw-bold"
        >
          childFriendly
        </div>
      </div>
      <span
        class="badge rounded-pill bg-primary"
      >
        3
      </span>
    </li>
    <li
      class="d-flex justify-content-between align-items-start list-group-item"
    >
      <div
        class="ms-2 me-auto"
      >
        <div
          class="fw-bold"
        >
          grooming
        </div>
      </div>
      <span
        class="badge rounded-pill bg-primary"
      >
        4
      </span>
    </li>
    <li
      class="d-flex justify-content-between align-items-start list-group-item"
    >
      <div
        class="ms-2 me-auto"
      >
        <div
          class="fw-bold"
        >
          intelligence
        </div>
      </div>
      <span
        class="badge rounded-pill bg-primary"
      >
        5
      </span>
    </li>
    <li
      class="d-flex justify-content-between align-items-start list-group-item"
    >
      <div
        class="ms-2 me-auto"
      >
        <div
          class="fw-bold"
        >
          healthIssues
        </div>
      </div>
      <span
        class="badge rounded-pill bg-primary"
      >
        6
      </span>
    </li>
    <li
      class="d-flex justify-content-between align-items-start list-group-item"
    >
      <div
        class="ms-2 me-auto"
      >
        <div
          class="fw-bold"
        >
          socialNeeds
        </div>
      </div>
      <span
        class="badge rounded-pill bg-primary"
      >
        7
      </span>
    </li>
    <li
      class="d-flex justify-content-between align-items-start list-group-item"
    >
      <div
        class="ms-2 me-auto"
      >
        <div
          class="fw-bold"
        >
          strangerFriendly
        </div>
      </div>
      <span
        class="badge rounded-pill bg-primary"
      >
        8
      </span>
    </li>
  </ol>
</div>
`;
